python/src/ir.cc

```diff
@@ -500,6 +510,54 @@ void init_triton_ir(py::module &&m) {
            [](ModuleOp &self, std::string &funcName) -> FuncOp {
              return self.lookupSymbol<FuncOp>(funcName);
            })
+      /*
+       * def ty_to_cpp(ty) is the consumer of this function.
+       * If the type is a ptr it expects ty[0] == '*', else the type itself.
+       */
+
+      .def("get_function_signature",
+           [](ModuleOp &self, FuncOp &func) -> std::vector<std::string> {
+             std::vector<std::string> strVec;
+
+             auto findTMA = [](ArrayRef<NamedAttribute> dictVals) {
+               for (auto attr : dictVals) {
+                 if (auto intAttr = dyn_cast<IntegerAttr>(attr.getValue())) {
```

Comment:
This doesn't seem to check that the name of the attribute is actually `nv_tma_desc`? It only checks that the value is 1?

Reply:
should be fixed